File: src/Python/901-1000/905.SortArrayByParity.py

class Solution:
    def sortArrayByParity(self, A):
        """
        :type A: List[int]
        :rtype: List[int]
        """
        left = 0
        right = len(A) - 1
        while left < right:
            if A[left] % 2 == 0:
                left += 1
            else:
                if A[right] % 2 == 0:
                    A[left], A[right] = A[right], A[left]
                right -= 1
        return A


if __name__ == '__main__':
    solution = Solution()
    print(solution.sortArrayByParity([3, 1, 2, 4]))
    print(solution.sortArrayByParity([3, 1, 2, 4, 5, 6]))
else:
    pass

File: src/Java/901-1000/905.SortArrayByParity.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int[] sortArrayByParity(int[] A) {
        int left = 0;
        int right = A.length - 1;
        while (left < right){
            if (A[left] % 2 == 0)
                left += 1;
            else {
                if (A[right] % 2 == 0){
                    int temp = A[left];
                    A[left] = A[right];
                    A[right] = temp;
                }
                right -= 1;
            }
        }
        return A;
    }
}
